File: app/src/main/java/home/oleg/placesnearme/di/modules/DbModule.java
package home.oleg.placesnearme.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import home.oleg.placesnearme.DataBaseFactory;
import home.oleg.placesnearme.dao.DetailedVenueWithPhotosDao;
import home.oleg.placesnearme.database.AppDatabase;
import home.oleg.placesnearme.database.DaoProvider;

@Module
public class DbModule {

    @Singleton
    @Provides
    static DaoProvider provideDaoProvider(Context context) {
        return DataBaseFactory.create(context);
    }

    @Singleton
    @Provides
    static DetailedVenueWithPhotosDao provideDetailedVenueWithPhotosDao(DaoProvider provider) {
        return provider.getDetailedVenueWithPhotosDao();
    }


}

File: app/src/main/java/home/oleg/placesnearme/app/di/modules/DbModule.kt
package home.oleg.placesnearme.app.di.modules

import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import home.oleg.placesnearme.data.DaoProviderFactory
import home.oleg.placesnearme.data.dao.DetailedVenueDao
import home.oleg.placesnearme.data.dao.DetailedVenueHistoryDao
import home.oleg.placesnearme.data.database.DaoProvider

@Module
object DbModule {

    @JvmStatic
    @Singleton
    @Provides
    fun provideDaoProvider(context: Context): DaoProvider {
        return DaoProviderFactory.create(context)
    }

    @JvmStatic
    @Singleton
    @Provides
    fun provideDetailedVenueWithPhotosDao(provider: DaoProvider): DetailedVenueDao {
        return provider.detailedVenueWithPhotosDao
    }

    @JvmStatic
    @Singleton
    @Provides
    fun provideDetailedVenueHistoryDao(provider: DaoProvider): DetailedVenueHistoryDao {
        return provider.venueHistoryDao
    }

}
